plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -626,7 +705,31 @@ enterComputeCek costs = computeCek where
       case resultOrErr of
           -- Turn the cause of a possible failure, being a 'CekValue', into a 'Term'.
           Left e       -> throwCek $ mapCauseInMachineException (void . dischargeCekValue) e
-          Right result -> returnCek ctx result
+          Right result -> returnCek unbudgetedSteps ctx result
+
+    -- | Spend the budget that has been accumulated for a number of machine steps.
+    spendAccumulatedBudget :: WordArray -> CekM s ()
+    spendAccumulatedBudget !unbudgetedSteps = iforWordArray unbudgetedSteps spend
+
+    -- Making this a definition of its own causes it to inline better than actually writing it inline, for
+    -- some reason.
+    -- Skip index 7, that's the total counter!
+    -- See Note [Structure of the step counter]
+    {-# INLINE spend #-}
+    spend !i !w = unless (i == 7) $ do
```

Comment:
Isn't `i` already forced anyway? And even if it's not, you're forcing it right on the same line via `i == 7` anyway. I don't think the bang is needed.

Reply:
I guess I'm following a cruder heuristic: I want this to be a "strict function" so I'm just going to bang all the arguments even if maybe they don't need it. I don't think that *hurts*?